server/src/Service/InteractionService.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace App\Service;
+
+use App\Dao\RevisionDao;
+
+class InteractionService {
+
+	/**
+	 * @var RevisionDao
+	 */
+	private $revisionDao;
+
+	/**
+	 * @param RevisionDao $revisionDao
+	 */
+	public function __construct( RevisionDao $revisionDao ) {
+		$this->revisionDao = $revisionDao;
+	}
+
+	/**
+	 * @param string[] $users
+	 * @param int|null $startDate
+	 * @param int|null $endDate
+	 * @param int $limit
+	 * @param string|null $continue
+	 * @return array
+	 */
+	public function getInteraction(
+		$users, $startDate = null, $endDate = null, $limit = 50, $continue = null
+	) {
+		$this->validateLimit( $limit );
+
+		list( $revisionIds, $continue ) = $this->revisionDao->getUserRevisionsInCommonPages(
+				$users,  $startDate, $endDate, $limit, $continue
+		);
+
+		$revisions = $this->revisionDao->getRevisionDetails( $revisionIds );
+		$revisions = $this->parseAndMapRevisions( $revisions );
+
+		return [ $revisions, $continue ];
+	}
+
+	/**
+	 * @param int[] $revisions
+	 * @return array
+	 */
+	private function parseAndMapRevisions( $revisions ) {
+		$interaction = [];
+		foreach ( $revisions as $revision ) {
+			$revision = [
+				'rev_id' => (int)$revision['rev_id'],
+				'page_id' => (int)$revision['rev_page'],
+				'page_title' => $revision['page_title'],
+				'page_namespace' => (int)$revision['page_namespace'],
+				'user' => $revision['rev_user_text'],
+				'timestamp' => strtotime( $revision['rev_timestamp'] ),
```

Comment:
You can leave this as whatever is in the database. No reason to convert it to a unix timestamp since I'm going to throw it into moment anyways.

Reply:
db format is weird, Ex: '20180227130400'. If would rather have it as a timestamp if you don't mind.

Reply:
ooo, I was thinking it was similar to what the existing API returns.. ok, nvm.